### DISCUSSION TITLE
Building a multi-block plugin

### DISCUSSION THREAD

**Discussion by troychaplin** [WordPress/developer-blog-content#289] [Topic Ideas]
Create block is a fantastic to help get you up and running with a new block rather quickly. but what if you plan to build multiple blocks and want to keep them all contained within one plugin?

I was faced with this challenge in a recent project and think sharing my experience and configuration would benefit other developers. This post would include plenty of code examples as well as a repo with a multi-block starter project that I would build during the process of writing the post.

**Comment by troychaplin** [comment]
This post would take a different approach that [what was written here](https://developer.wordpress.org/news/2024/05/22/setting-up-a-multi-block-using-inner-blocks-and-post-meta/), which would be linked up in my post in reference to setting it up as a parent with a single innerblock.

**Reply by bph** [reply]
Hi @troychaplin! Thank you for submitting yet another topic idea for the Develop Blog. So glad to have you. 🎉

Would you mind outlining the "different approach" some more? A [fat outline ](https://bernoff.com/blog/fat-outline) would certainly help. Doesn't have to be all too detailed. It should answer the question, what specific challenges you faced, how you overcame them and what a reader would learn.

**Reply by troychaplin** [reply]
Gladly. Just gotta make a bit of time to put that outline together.

**Reply by troychaplin** [reply]
Hey @bph, finally getting some time to think this out more. Here's what my initial idea was for this post:

### Introduction
- Brief overview of the concept and what a multi-block plugin is
- Provide idea of what reader will build by the end

### Benefits
- Less maintenance, easier dependency management
- Increase shared code, reduce duplication across pugins
- Potential improvements to security response time (I'm no expert but a hole in a key dependency is easier to address and contain in 1 plugin than 10, thinking of posing a question to readers to encourage commenting and not making a factual statement about this one)

### Basic Setup
- Brief overview of using @wordpress/create-block
- Refactor default folder structure
- How to add a second block

### Configuration Improvements
- Handling multiple block registrations in one function
- Add webpack to output all blocks in one js file
- Example of adding javascript for other purposes (such as slot fills, block variation, more)

### Wrap Up
- Summary of key points
- Provide links to related WP maintained docs

### Bonus Section: Integrating TailwindCSS
- Overview of what Tailwind is and it's benefits
- Setting up Tailwind in your multi-block plugin
- Using Tailwind in your blocks

**Comment by zzap** [comment]
@troychaplin please ping me when draft is ready for the first review. Thanks

**Comment by bph** [comment]
This topic was approved. Please continue conversation on the issue: 
- #292

**Comment by troychaplin** [comment]
Article has been published https://developer.wordpress.org/news/2024/09/17/how-to-build-a-multi-block-plugin/
